src/include/duckdb/common/types/string.hpp

```diff
@@ -0,0 +1,190 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/common/types/string.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/common/numeric_utils.hpp"
+#include "duckdb/common/typedefs.hpp"
+
+namespace duckdb {
+class String {
+public:
+	// Owning constructors
+	String(std::string str) // NOLINT: allow implicit conversion
+	    : owned_data(std::move(str)), data(owned_data.c_str()), size(owned_data.size()) {
+	}
+
+	String(const char *ptr, const size_t len) : owned_data(ptr, len), data(ptr), size(len) {
+	}
+
+	String(const char *ptr) // NOLINT: allow implicit conversion
+	    : String(ptr, strlen(ptr)) {
+	}
+
+	// Non-owning constructors
+	static String CreateView(const char *ptr, const size_t len) {
```

Comment:
Maybe a bit of a nit, but if we do add a StringView in the future this is a bit confusing, maybe we can call it something like `Reference()` or `NonOwning()`